optimistic-concurrency/src/main/java/com/iluwatar/optimisticconcurrency/App.java

```diff
@@ -0,0 +1,107 @@
+package com.iluwatar.optimisticconcurrency;
+
+import java.util.Optional;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.Persistence;
+
+import lombok.extern.slf4j.Slf4j;
+
+
+/**
+ * Optimistic Concurrency pattern.
+ * Increment version during every update.
+ * Attempts to update object using old version fail
+ */
```

Comment:
Please add more verbose explanation what the pattern is and how this example implements it